### ISSUE TITLE
[BUG] Pagerfanta Outputwalker breaks sorting in doctrine repository

### ISSUE THREAD

**Issue by koemeet** [Sylius/Sylius#3268]
The PR #3185 is causing a lot of problems for me. Whevener I apply sorting to the repository, the whole SQL query is wrong. This is the error I am getting:

``` sql
An exception occurred while executing 
    'SELECT DISTINCT id_4 
        FROM (
                SELECT m0_.available_on AS available_on_0, m0_.created_at AS created_at_1, 
                    m0_.updated_at AS updated_at_2, m0_.deleted_at AS deleted_at_3, m0_.id AS id_4, 
                    m0_.variant_selection_method AS variant_selection_method_5, m1_.name AS name_6, 
                    m1_.slug AS slug_7, m1_.description AS description_8, m1_.meta_keywords AS meta_keywords_9, 
                    m1_.meta_description AS meta_description_10, m1_.id AS id_11, 
                    m1_.short_description AS short_description_12, m1_.locale AS locale_13,
                    m2_.is_master AS is_master_14, m2_.presentation AS presentation_15, m2_.created_at AS created_at_16, 
                    m2_.updated_at AS updated_at_17, m2_.deleted_at AS deleted_at_18, m2_.id AS id_19, 
                    m2_.available_on AS available_on_20, m2_.sku AS sku_21, m2_.on_hold AS on_hold_22, m2_.on_hand AS on_hand_23, 
                    m2_.sold AS sold_24, m2_.available_on_demand AS available_on_demand_25, m2_.price AS price_26, 
                    m2_.pricing_calculator AS pricing_calculator_27, m2_.pricing_configuration AS pricing_configuration_28, 
                    m2_.width AS width_29, m2_.height AS height_30, m2_.depth AS depth_31, m2_.weight AS weight_32 
                FROM mango_product m0_ LEFT JOIN mango_product_translation m1_ ON m0_.id = m1_.translatable_id LEFT JOIN mango_product_variant m2_ ON m0_.id = m2_.product_id
            ) dctrn_result 
        ORDER BY m0_.updated_at DESC 
        LIMIT 10 OFFSET 0':\n\nSQLSTATE[42S22]: Column not found: 1054 Unknown column 'm0_.updated_at' in 'order clause'
```

I use the `resource-bundle` and the `product-bundle` and have setup a route exactly like the `core-bundle` does:

``` yaml
mango_api_product_index:
    path: /products
    methods: [GET]
    defaults:
        _controller: sylius.controller.product:indexAction
        _sylius:
            sortable: true
            repository:
                method: createFilterPaginator # I have also copied this code from the corebundle to test it
                arguments: [$criteria, $sorting, $deleted]
```

When I revert the commit from PR #3185 everyting works perfectly fine. Am I missing something or is this really a bug?

**NOTE: I do not use the Sylius standard, but only the bundles, maybe there is something in CoreBundle that I do not have?**

**Comment by steffenbrem** [comment]
I can confirm that the cause is `doctrine/orm` version `2.5` and above. To fix my issue, I've put the following in my composer.json:
```json
 "doctrine/orm": ">=2.4.8,<2.5"
```

@pjedrzejewski Can I close this? Not sure if my issue is relevant to Sylius.

**Comment by michalmarcinkowski** [comment]
The issue is with Doctrine, that's why we have this https://github.com/Sylius/Sylius/blob/master/composer.json#L30 requirement.

The problem occurs for sure in ProductBundle, @pjedrzejewski should we add this doctrine restriction to ProductBundle's composer.json?

**Comment by bigfoot90** [comment]
cc @bigfoot90

**Comment by pjedrzejewski** [comment]
@pamil Could you remind me about the bug in Doctrine 2.5.0? I think we needed to use 2.5.1?

**Comment by lighta971** [comment]
I did a fresh install of Sylius 0.15 and I have this problem too in Backend product-options and product-archetypes index pages.

```
An exception has been thrown during the rendering of a template ("An exception occurred while executing 'SELECT DISTINCT id_3 FROM (SELECT s0_.code AS code_0, s0_.created_at AS created_at_1, s0_.updated_at AS updated_at_2, s0_.id AS id_3, s1_.name AS name_4, s1_.id AS id_5, s1_.locale AS locale_6 FROM sylius_product_archetype s0_ LEFT JOIN sylius_product_archetype_translation s1_ ON s0_.id = s1_.translatable_id) dctrn_result ORDER BY s0_.code DESC LIMIT 10 OFFSET 0':

SQLSTATE[42S22]: Column not found: 1054 Unknown column 's0_.code' in 'order clause'") in SyliusWebBundle:Backend/ProductArchetype:index.html.twig at line 25.
```

**Comment by pjedrzejewski** [comment]
@lighta971 Did you install Sylius/Sylius or Sylius/Sylius-Standard?

**Comment by lighta971** [comment]
The command I ran is:
`composer create-project -s dev sylius/sylius-standard:0.15.0 acme`

In my composer.json
` "sylius/sylius":                "0.15.*@dev"`

**Comment by michalmarcinkowski** [comment]
It's Doctrine issue, you need to add https://github.com/Sylius/Sylius/issues/3268#issuecomment-139600135 to your composer.json. We're going to add this to Sylius-Standard.

**Comment by axelvnk** [comment]
Had the same problem. I can confirm that the issue is resolved by downgrading to Doctrine 2.4

**Comment by rvanlaak** [comment]
see https://github.com/Sylius/Sylius/pull/3684#issuecomment-218472227
